x/pairing/types/scores/stake_req.go

```diff
@@ -0,0 +1,16 @@
+package score
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	epochstoragetypes "github.com/lavanet/lava/x/epochstorage/types"
+)
+
+// stake requirement that implements the ScoreReq interface
+type StakeReq struct {
+	stake sdk.Coin
+}
+
+// calculates the stake score of a provider (which is simply the normalized stake)
+func (sr StakeReq) Score(provider epochstoragetypes.StakeEntry, weight uint64) uint64 {
+	return provider.Stake.Amount.ToDec().Power(weight).BigInt().Uint64()
```

Comment:
outdated. in the newer version, I normalize the stake to avoid overflow

Reply:
what do you mean?
the weight is already uint64 and the provider's stake is saved as a sdk.Coin object which I can't change (we use it because it's more secure than uint64)